frontend/src/modules/Pvt/utils/PvtPlotBuilder.ts

```diff
@@ -0,0 +1,335 @@
+                            ) {
+                                continue;
+                            }
+
+                            let dependentVariableValue = 0;
+                            switch (dependentVariable) {
+                                case PressureDependentVariable.FORMATION_VOLUME_FACTOR:
+                                    dependentVariableValue = table.volumefactor[i];
+                                    break;
+                                case PressureDependentVariable.VISCOSITY:
+                                    dependentVariableValue = table.viscosity[i];
+                                    break;
+                                case PressureDependentVariable.DENSITY:
+                                    dependentVariableValue = table.density[i];
+                                    break;
+                                case PressureDependentVariable.FLUID_RATIO:
+                                    dependentVariableValue = table.ratio[i];
+                                    break;
+                            }
+
+                            let dependentVariableMap = groupedTracesMaps.get(dependentVariable);
+                            if (!dependentVariableMap) {
+                                dependentVariableMap = new Map<number, TracePointData[]>();
+                                groupedTracesMaps.set(dependentVariable, dependentVariableMap);
+                            }
+
+                            let groupValue = 0;
+                            if (phase === PhaseType.OIL) {
+                                groupValue = table.ratio[i];
+                            } else if (phase === PhaseType.GAS) {
+                                groupValue = table.pressure[i];
+                            }
+
+                            let tracePointDataArray = dependentVariableMap.get(groupValue);
+                            if (tracePointDataArray === undefined) {
+                                tracePointDataArray = [];
+                                dependentVariableMap.set(groupValue, tracePointDataArray);
+                            }
+                            if (dependentVariable === PressureDependentVariable.FLUID_RATIO) {
+                                if (tracePointDataArray.length > 0) {
+                                    continue;
+                                }
+                            }
+                            tracePointDataArray.push({
+                                ratio: table.ratio[i],
+                                pressure: table.pressure[i],
+                                dependentVariableValue,
+                            });
+                        }
+                    }
+
+                    let i = 0;
+                    let color = colors[collectionIndex];
+                    if (colorBy === ColorBy.PVT_NUM) {
+                        color = colors[pvtNumIndex];
+                    }
```

Comment:
Ensure valid indices?

Reply:
The color array is created with exactly the required amount of colors, see line 137. Hence, IMO this check is not required. If the index is invalid, the usual exception should be explanation enough.